pkg/controllers/node/health/controller.go

```diff
@@ -73,7 +75,29 @@ func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
 	log.FromContext(ctx).V(1).Info("node.health maxConcurrentReconciles set", "maxConcurrentReconciles", maxConcurrentReconciles)
 	return controllerruntime.NewControllerManagedBy(m).
 		Named("node.health").
-		For(&corev1.Node{}, builder.WithPredicates(nodeutils.IsManagedPredicateFuncs(c.cloudProvider))).
+		For(&corev1.Node{}, builder.WithPredicates(nodeutils.IsManagedPredicateFuncs(c.cloudProvider), predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldNode := e.ObjectOld.(*corev1.Node)
+				newNode := e.ObjectNew.(*corev1.Node)
+				if len(oldNode.Status.Conditions) != len(newNode.Status.Conditions) {
+					return true
+				}
+
+				for _, oldCond := range oldNode.Status.Conditions {
+					newCond := nodeutils.GetCondition(newNode, oldCond.Type)
+					// Return true if any of these conditions are met:
+					// 1. Condition type no longer exists in new node
+					// 2. Transition time has changed
+					// 3. Status has changed
+					if newCond.Type == "" ||
+						oldCond.LastTransitionTime != newCond.LastTransitionTime ||
```

Comment:
Just spot-checking: The heartbeat doesn't update the lastTransitionTime, right? That's just lastHearbeatTime?